doc/user_guide/pre-commit-integration.rst

```diff
@@ -0,0 +1,41 @@
+.. _pre-commit-integration:
+
+Pre-commit integration
+======================
+
+``pylint`` can be used as a `pre-commit <https://pre-commit.com>`_ hook.
+Since ``pylint`` needs to import modules and dependencies to work correctly, the
+hook only work with a local installation of ``pylint`` (in your environmnet).
+If you installed ``pylint`` locally it can be added to ``.pre-commit-config.yaml``
+as follows:
+
+.. sourcecode:: yaml
+
+  - repo: local
+    hooks:
+      - id: pylint
+        name: pylint
+        entry: pylint
+        language: system
+        types: [python]
+
+A hook with more configurations could look something like this:
+
+.. sourcecode:: yaml
+
+  - repo: local
+    hooks:
+      - id: pylint
+        name: pylint
+        entry: pylint
+        language: system
+        types: [python]
+        args:
+          [
+            "-rn", # Only display messages
+            "-sn", # Don't display the score
```

Comment:
``rn`` and ``sn`` are pretty much required, because pre-commit launch jobs in parallels so there will be dozen of score (or worst reports) if you don't. So I think we should put them in the simple example too.